Util.h

```diff
@@ -142,7 +142,8 @@ struct RegionRequest {
         config = config_;
     }
     bool IsAmong(int profile_index, std::vector<int> other_stats) {
-        if (config.size() > profile_index) {
+        // check dose the region spectral requirement with respect to the profile index exist
+        if ((config.size() > 0) && (profile_index >= 0) && (config.size() - 1 >= profile_index)) {
```

Comment:
The profile index can change.  If a user sets region 1 profile 0 = "z"/Sum, then adds a second zprofile widget for region 1 with "Qz"/Mean, the profiles for region 1 are now profile 0="Qz"/Mean and profile 1="z"/Sum.

Reply:
I'm refactoring the spectral profile cancellation according to your suggestions. The struct `RegionRequest` will be removed. Using the spectral profile configs which already stored in the `Region` to do the interruption checks.

Reply:
I would suggest fixing the hdf5 mask issue in this PR and refactor the interrupts separately.